File: HeartForCharity/HeartForCharity.Services/Database/Skill.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HeartForCharity.Services.Database
{
    public class Skill
    {
        [Key]
        public int SkillId { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = null!;

        [MaxLength(500)]
        public string? Description { get; set; }

        public virtual ICollection<VolunteerSkill> VolunteerSkills { get; set; } = new List<VolunteerSkill>();
    }
}

File: HeartForCharity/UI/heartforcharity_mobile/lib/model/responses/skill.dart
class Skill {
  final int skillId;
  final String name;
  final String? description;

  Skill({required this.skillId, required this.name, this.description});

  factory Skill.fromJson(Map<String, dynamic> json) => Skill(
        skillId: json['skillId'] ?? 0,
        name: json['name'] ?? '',
        description: json['description'],
      );
}
